include/common/async.h

```diff
@@ -66,20 +65,15 @@ template <typename T> class Async {
     using std::swap;
     swap(LHS.Future, RHS.Future);
     swap(LHS.Thread, RHS.Thread);
-    swap(LHS.VMPtr, RHS.VMPtr);
+    swap(LHS.StopFuncPtr, RHS.StopFuncPtr);
   }
 
-  void cancel() noexcept {
-    if (likely(VMPtr)) {
-      VMPtr->stop();
-    }
-  }
+  void cancel() noexcept { StopFuncPtr(); }
```

Comment:
```c++
void cancel() noexcept {
  if (likely(StopFuncPtr)) {
    StopFuncPtr();
  }
}
```

Reply:
The `std::function` cannot support this.

Reply:
Use `operator bool()` instead.